### PR TITLE
add samurai package. 

### PR DESCRIPTION
**Pull request by vtorri**
samurai is a ninja-compatible build tool written in C

### CODE REVIEW

File: components/developer/samurai/manifests/sample-manifest.p5m [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+set name=pkg.human-version value=$(HUMAN_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+file path=export/home/vtorri/gitroot/oi-userland/components/developer/samurai/build/prototype/$(MACH)/usr/bin/samu
```
Comment (Toasterson) [inline components/developer/samurai/manifests/sample-manifest.p5m]: This is wrong it should be `file path=usr/bin/samu` the autodetection looks broken on your system

**Review by Toasterson — CHANGES_REQUESTED** [review]
Please fix your autodetection of paths they must be relative to the prototype directory.

**Review by Toasterson — CHANGES_REQUESTED** [review]
paths must be relative. Please use `gmake sample-manifest` as example.

File: components/developer/samurai/Makefile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+COMPONENT_ARCHIVE=        $(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_URL=    https://git.sr.ht/~mcf/samurai/archive/$(COMPONENT_VERSION).tar.gz
+COMPONENT_ARCHIVE_HASH=   sha256:37a2d9f35f338c53387eba210bab7e5d8abe033492664984704ad84f91b71bac
+COMPONENT_FMRI=           developer/build/samurai
+COMPONENT_CLASSIFICATION= Development/Distribution Tools
+COMPONENT_LICENSE=        Apache 2.0
+
+include $(WS_MAKE_RULES)/common.mk
+
+COMPONENT_BUILD_ENV += CC="/usr/bin/gcc"
```
Comment (alarcher) [inline components/developer/samurai/Makefile:34]: You could even just pass CC=$(CC) to use the compiler toolchain selected with GCC_VERSION.

**Comment by vtorri** [comment]
@alarcher thank you, i'll redo it with @Toasterson's patch:
https://gist.github.com/Toasterson/d937cb678fccb2dabae8cb5de9450a40

**Comment by vtorri** [comment]
I've updated my gist:

https://gist.github.com/vtorri/d1d78062795832403e334a132bad702f

I'll try it this evening or tomorrow

**Comment by vtorri** [comment]
will do another PR, closing this one

**Pull request closed without merging**